File: src/main/java/me/ag2s/epublib/domain/FileResourceProvider.java
package me.ag2s.epublib.domain;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class FileResourceProvider implements LazyResourceProvider {
   String dir;

   public FileResourceProvider(String parentDir) {
      this.dir = parentDir;
   }

   public FileResourceProvider(File parentFile) {
      this.dir = parentFile.getPath();
   }

   public InputStream getResourceStream(String href) throws IOException {
      return new FileInputStream(new File(this.dir, href));
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/FileResourceProvider.kt
package me.ag2s.epublib.domain

import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream

class FileResourceProvider : LazyResourceProvider {
    internal var dir: String
    constructor(parentDir: String) { dir = parentDir }
    constructor(parentFile: File) { dir = parentFile.path }
    @Throws(IOException::class)
    override fun getResourceStream(href: String?): InputStream = FileInputStream(File(dir, href))
}
